File: src/net/auoeke/cin/element/IntegerElement.java
package net.auoeke.cin.element;

public class IntegerElement extends NumberElement {
    public long value;

    public IntegerElement(long value) {
        this.value = value;
    }

    @Override public Type type() {
        return Type.INTEGER;
    }

    @Override public int intValue() {
        return (int) this.value;
    }

    @Override public long longValue() {
        return this.value;
    }

    @Override public float floatValue() {
        return this.value;
    }

    @Override public double doubleValue() {
        return this.value;
    }

    @Override public byte byteValue() {
        return (byte) this.value;
    }

    @Override public short shortValue() {
        return (short) this.value;
    }

    @Override public int hashCode() {
        return Double.hashCode(this.value);
    }

    @Override public boolean equals(Object other) {
        return other instanceof IntegerElement integer && this.value == integer.value;
    }

    @Override public String toString() {
        return Long.toString(this.value);
    }
}

File: src/net/auoeke/cin/element/IntegerElement.kt
package net.auoeke.cin.element

class IntegerElement(var value: Long) : NumberElement() {
    override val type: ElementType get() = ElementType.INTEGER

    override fun toByte(): Byte = value.toByte()
    override fun toChar(): Char = value.toInt().toChar()
    override fun toDouble(): Double = value.toDouble()
    override fun toFloat(): Float = value.toFloat()
    override fun toInt(): Int = value.toInt()
    override fun toLong(): Long = value
    override fun toShort(): Short = value.toShort()
    override fun toString(): String = value.toString()
}
